### ISSUE TITLE
Model assumes identical image shape

### ISSUE THREAD

**Issue by vitasiku** [experiencor/basic-yolo-keras#78]
Thanks for your work. However, the design of the model assumes identical dimensions for width, height. For instance, in the frontend.py file the Yolo model, receives a single argument of input_size which is passed both as height and width in the line 

`input_image     = Input(shape=(self.input_size, self.input_size, 3))`

Therefore, this breaks when supplied with different width and height for the image.

**Comment by experiencor** [comment]
@vitasiku you are right. That's why the generator always resizes the images to the same dimension.

**Comment by vitasiku** [comment]
@experiencor, given a training set made up of 1024x768 images that have been annotated whereby the annotation was done at their original size. Does the generator also resize the annotation of the objects within the images? Or does that require reannotating the images at identical image shapes?

**Comment by experiencor** [comment]
Yes the generator also resizes annotations. You don't need to reannotate.

**Comment by vitasiku** [comment]
Thanks for the clarification. But how about the frontend.py receives the original image dimensions even though not identical. Otherwise, how would one specify the original dimensions(in this case 1024x768) since the model only takes one dimension?  
Also,
I have a dataset where the annotated objects are very tiny compared to the rest of the image and am getting **0 recall** at the moment though the loss keeps on changing. Perhaps, the resizing worsens the problem of detecting the already small objects. 

Do you have any suggestion for the case of dealing with small objects compared to the rest of the images?

**Comment by experiencor** [comment]
@vitasiku You don't have to specify the original dimensions. All dimensions are resized to (input_size, input_size).

You may want to change the anchor sizes. It's defaulted to anchors based on VOC dataset. Your dataset is very different from VOC dataset, hence changing anchors is a must.

**Comment by vitasiku** [comment]
Thanks. I will try that.
